c/sedona-geos/src/st_exteriorring.rs

```diff
@@ -0,0 +1,165 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::executor::GeosExecutor;
+use arrow_array::builder::BinaryBuilder;
+use datafusion_common::{error::Result, DataFusionError};
+use datafusion_expr::ColumnarValue;
+use geos::{Geom, GeometryTypes::Polygon, OutputDimension, WKBWriter};
+use sedona_expr::scalar_udf::{ScalarKernelRef, SedonaScalarKernel};
+use sedona_schema::{
+    datatypes::{SedonaType, WKB_GEOMETRY},
+    matchers::ArgMatcher,
+};
+use std::sync::Arc;
+
+pub fn st_exterior_ring_impl() -> ScalarKernelRef {
+    Arc::new(STExteriorRing {})
+}
+
+#[derive(Debug)]
+struct STExteriorRing {}
+
+impl SedonaScalarKernel for STExteriorRing {
+    fn return_type(&self, args: &[SedonaType]) -> Result<Option<SedonaType>> {
+        let matcher = ArgMatcher::new(vec![ArgMatcher::is_geometry()], WKB_GEOMETRY);
+        matcher.match_args(args)
+    }
+
+    fn invoke_batch(
+        &self,
+        arg_types: &[SedonaType],
+        args: &[ColumnarValue],
+    ) -> Result<ColumnarValue> {
+        let executor = GeosExecutor::new(arg_types, args);
+        let mut builder = BinaryBuilder::with_capacity(
+            executor.num_iterations(),
+            100 * executor.num_iterations(),
```

Comment:
```suggestion
            WKB_MIN_PROBABLE_BYTES * executor.num_iterations(),
```

To figure out how to set values like this, I recommend looking at other functions that return a geometry like `st_centroid.rs`. e.g. [here](https://github.com/apache/sedona-db/blob/c48dfaa06858b5fe3ddf8d68115a890fe222cab1/c/sedona-geos/src/st_centroid.rs#L53-L56)

You'll need to add the import at the top too:
```rust
use sedona_geometry::wkb_factory::WKB_MIN_PROBABLE_BYTES;
```